### PR TITLE
fix(tableau): retry on InternalServerError 504

### PR DESCRIPTION
**Pull request by sgomezvillamor**
Retries on the following case

```
Internal error 504 at https://prod-useast-a.online.tableau.com/api/metadata/graphql
b''
Traceback (most recent call last):
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/run/pipeline.py", line 465, in run
    for wu in itertools.islice(
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/source/state/stale_entity_removal_handler.py", line 72, in auto_stale_entity_removal
    for wu in stream:
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/api/auto_work_units/auto_dataset_properties_aspect.py", line 62, in auto_patch_last_modified
    for wu in stream:
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/api/source_helpers.py", line 149, in auto_workunit_reporter
    for wu in stream:
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/api/source_helpers.py", line 253, in auto_browse_path_v2
    for urn, batch in _batch_workunits_by_urn(stream):
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/api/source_helpers.py", line 512, in _batch_workunits_by_urn
    for wu in stream:
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/api/source_helpers.py", line 416, in auto_fix_empty_field_paths
    for wu in stream:
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/api/source_helpers.py", line 368, in auto_fix_duplicate_schema_field_paths
    for wu in stream:
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/api/source_helpers.py", line 172, in auto_materialize_referenced_tags_terms
    for wu in stream:
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/api/source_helpers.py", line 104, in auto_status_aspect
    for wu in stream:
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/source/tableau/tableau.py", line 798, in get_workunits_internal
    yield from site_source.ingest_tableau_site()
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/source/tableau/tableau.py", line 3437, in ingest_tableau_site
    yield from self.emit_embedded_datasources()
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/source/tableau/tableau.py", line 3250, in emit_embedded_datasources
    datasource = self.update_datasource_for_field_upstream(
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/source/tableau/tableau.py", line 2538, in update_datasource_for_field_upstream
    for field_upstream in self.get_connection_objects(
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/source/tableau/tableau.py", line 1342, in get_connection_objects
    ) = self.get_connection_object_page(
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/source/tableau/tableau.py", line 1173, in get_connection_object_page
    query_data = query_metadata_cursor_based_pagination(
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/datahub/ingestion/source/tableau/tableau_common.py", line 959, in query_metadata_cursor_based_pagination
    result = server.metadata.query(
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/tableauserverclient/server/endpoint/endpoint.py", line 274, in wrapper
    return func(self, *args, **kwargs)
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/tableauserverclient/server/endpoint/metadata_endpoint.py", line 71, in query
    server_response = self.post_request(url, graphql_query, content_type="application/json", parameters=parameters)
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/tableauserverclient/server/endpoint/endpoint.py", line 226, in post_request
    return self._make_request(
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/tableauserverclient/server/endpoint/endpoint.py", line 141, in _make_request
    self._check_status(server_response, url)
  File "/tmp/datahub/ingest/venv-tableau-ab9d9cb03b44c5de/lib/python3.10/site-packages/tableauserverclient/server/endpoint/endpoint.py", line 159, in _check_status
    raise InternalServerError(server_response, url)
tableauserverclient.server.endpoint.exceptions.InternalServerError: 

Internal error 504 at https://prod-useast-a.online.tableau.com/api/metadata/graphql
b''
```

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py [hunk 1]
```diff
@@ -1196,6 +1199,22 @@ def get_connection_object_page(
...
+                    raise
+                return self.get_connection_object_page(
+                    query=query,
+                    connection_type=connection_type,
+                    query_filter=query_filter,
+                    fetch_size=fetch_size,
+                    current_cursor=current_cursor,
+                    retry_on_auth_error=False,
+                    retries_remaining=retries_remaining - 1,
+                )
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py:1216]: do we need an `else: raise`

File: metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py [hunk 2]
```diff
@@ -1196,6 +1199,22 @@ def get_connection_object_page(
...
+                if retries_remaining <= 0:
+                    raise
+                return self.get_connection_object_page(
+                    query=query,
+                    connection_type=connection_type,
+                    query_filter=query_filter,
+                    fetch_size=fetch_size,
+                    current_cursor=current_cursor,
+                    retry_on_auth_error=False,
+                    retries_remaining=retries_remaining - 1,
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py:1215]: not a huge fan of this entire method - it feels way more complicated than it needs to be and has a ton of mostly duplicated code

**Review by hsheth2 — APPROVED** [review]
Approving to unblock

File: metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py (hunk 1 above)
Comment (sgomezvillamor) [reply metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py:1216]: indeed

File: metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py (hunk 2 above)
Comment (sgomezvillamor) [reply metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py:1215]: I agree
If this works, we should invest some time on a refactor.

**Pull request merged**